ibis/expr/datatypes.py

```diff
@@ -195,45 +217,43 @@ class Timestamp(Primitive):
 
     __slots__ = 'timezone',
 
-    def __init__(self, timezone=None, nullable=True):
-        super(Timestamp, self).__init__(nullable=nullable)
+    def __init__(
+        self,
+        timezone: Optional[str] = None,
+        nullable: bool = True
+    ) -> None:
+        super().__init__(nullable=nullable)
         self.timezone = timezone
 
-    def _equal_part(self, other, cache=None):
-        return self.timezone == other.timezone
-
-    def __call__(self, timezone=None, nullable=True):
-        return type(self)(timezone=timezone, nullable=nullable)
-
-    def __str__(self):
+    def __str__(self) -> str:
         timezone = self.timezone
         typename = self.name.lower()
         if timezone is None:
             return typename
         return '{}({!r})'.format(typename, timezone)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
+        # Can't use super here because the parent method doesn't print the
```

Comment:
Perhaps it's not primitive then? How about adjusting the `DataType.__repr__` and `Primitive.__repr__` instead?

Reply:
Yeah that makes sense.